src/Microsoft.Health.Fhir.Anonymizer.Shared.Core/Processors/GeneralizeProcess.cs

```diff
@@ -0,0 +1,174 @@
+            {
+                node.Value = 0;
+            }
+            else
+            {
+                node.Value = targetIntValue;
+            }
+        }
+
+        private void GeneralizeDateOrTimerNode(ElementNode node, string targetExpression)
+        {
+            var targetValue = node.Scalar(targetExpression);
+            if (string.Equals(targetValue.GetType().ToString(), node.InstanceType))
+            {
+                node.Value = targetValue;
+                return;
+            }
+            try
+            {
+                if (string.Equals(FHIRAllTypes.Time.ToString(), node.InstanceType, StringComparison.InvariantCultureIgnoreCase))
+                {
+                    node.Value = PartialTime.Parse(targetValue.ToString());
+                }
+                else
+                {
+                    node.Value = PartialDateTime.Parse(targetValue.ToString());
+                }
+            }
+            catch (Exception ex)
+            {
+                throw new AnonymizerConfigurationErrorsException($"Invalid data types or format of  expression {targetExpression} output", ex);
+            }
+
+        }
+
+        private ProcessResult GeneralizeNode(ElementNode node, string targetExpression, ProcessResult result)
+        {
+            if (node.Scalar(targetExpression) == null)
+            {
+                node.Value = null;
+                result.AddProcessRecord(AnonymizationOperations.Generalize, node);
+                return result;
+            }
+
+            if (s_integerValueTypeNames.Contains(node.InstanceType, StringComparer.InvariantCultureIgnoreCase))
+            {
+                GeneralizeIntegerNode(node, targetExpression);
+            }
+            else if(s_DateOrTimeValueTypeNames.Contains(node.InstanceType, StringComparer.InvariantCultureIgnoreCase))
+            {
+                GeneralizeDateOrTimerNode(node, targetExpression);
+            }
+            else
+            {
+                node.Value = node.Scalar(targetExpression);
+            }
+
+            result.AddProcessRecord(AnonymizationOperations.Generalize, node);
+            return result;
+        }
+
+        public ProcessResult Process(ElementNode node, ProcessContext context = null, Dictionary<string, object> settings = null)
+        {
+
+            EnsureArg.IsNotNull(node);
+            EnsureArg.IsNotNull(context?.VisitedNodes);
+            EnsureArg.IsNotNull(settings);
+
+            var result = new ProcessResult();
+
+            if (!ModelInfo.IsPrimitive(node.InstanceType) || node.Value == null)
+            {
+                return result;
+            }
+            var generalizeSetting = GeneralizeSetting.CreateFromRuleSettings(settings);
+
+            JObject Cases = null;
+            try
+            {
+                Cases = JObject.Parse(generalizeSetting.Cases);
```

Comment:
Can we put the parsing logic in the setting file?

Reply:
Yes, the parsing process has been moved to setting file.